EVM-verifier/contracts/AliceRingToken.sol

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract RingSignatureVerifier {
+    constructor() {}
+
+    function verify(string memory signature) public pure returns (bool) {
+        return true;
+    }
+}
+
+contract AliceRingToken is ERC721, ERC721URIStorage, Ownable {
+    uint256 private _nextTokenId;
+
+    struct pubKey {
+        uint256 x;
+        uint256 y;
+    }
+
+    struct Signature {
+        pubKey[] ring;
+        uint256 c;
+        string message;
+        uint256[] responses;
+    }
+
+    enum Status {
+        UNKNOWN, // the proof has not been verified on-chain, no proof has been minted
+        MINTED // the proof has already been minted
+    }
+
+    error AlreadyMinted(string proofId);
+    error InvalidSignature();
+    error InvalidTokenAmounts();
+    error OnlyOwnerCanBurn(uint256 tokenId);
+
+    RingSignatureVerifier public verifier; // Instance of the ring signature verifier contract
+    uint256 public minimalAmount; // Minimal amount of tokens to be owned by each address in the ring

```

Comment:
This should not be a global variable. Each SBT proof can have a different token value. 
`minimalAmount` should be a function variable

Reply:
it should be an argument of `mint`. Our bad, we forgot it in the contract architecture